daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -47,29 +47,29 @@ case class DFDLXLeftShift(recipes: List[CompiledDPath], argType: NodeInfo.Kind)
     val width = argType.asInstanceOf[PrimNumeric].width.get
     Assert.invariant(shift >= 0)
     if(shift>=width)
-      dstate.SDE("Left shift not supported for shift greater or equal to "+ width)
+      dstate.SDE("Left shift not supported for shift greater or equal to %s for %s. Shift was %s", width, argType.globalQName, shift)
 
     argType match {
       case NodeInfo.Long => arg1.getLong << shift
       case NodeInfo.Int => arg1.getInt << shift
       case NodeInfo.Short => (arg1.getShort << shift).toShort
       case NodeInfo.Byte => (arg1.getByte << shift).toByte
-      //$COVERAGE-OFF$
       case NodeInfo.UnsignedLong => (ULong(arg1.getBigInt.longValue()) << shift).toBigInt
-      // $COVERAGE-ON$
       case NodeInfo.UnsignedInt => ULong(arg1.getLong << shift).toUInt.toLong
       case NodeInfo.UnsignedShort => UInt(arg1.getInt << shift).toUShort.toInt
       case NodeInfo.UnsignedByte => UInt(arg1.getShort << shift).toUByte.toShort
       //$COVERAGE-OFF$
-      case _ => Assert.invariantFailed("Left shift not supported")
+      case _ => Assert.invariantFailed(s"Left shift not supported for shift greater or equal to $width for ${argType.globalQName}. Shift was $shift")
```

Comment:
This new message isn't quite correct. If this case is it, it means someone passed in an unsupported type (e.g. xs:decimal), but somehow that wasn't caught in DFDLXShiftExpr. This error message maybe wants to say something like 

> Should not have gotten ${argType.globalQName} for left shift argument type"